### ISSUE TITLE
Warning in rails_best_practices hides actual result

### ISSUE THREAD

**Issue by waiting-for-dev** [sds/overcommit#428]
A warning in rails_best_practices hides actual code analysis warnings:

See the difference in the output:

```
➜  ONLY=RailsBestPractices overcommit -r

Running pre-commit hooks
Skipping YamlSyntax
Skipping RuboCop
Skipping RailsSchemaUpToDate
Skipping MergeConflicts
Skipping LocalPathsInGemfile
Skipping Reek
Skipping TrailingWhitespace
Skipping ExecutePermissions
Skipping CaseConflicts
Skipping BundleCheck
Skipping BundleAudit
Skipping BrokenSymlinks
Cannot skip AuthorName since it is required
Cannot skip AuthorEmail since it is required
Analyze with RailsBestPractices..................[RailsBestPractices] FAILED
(ripper):5: regexp encoding option 's' differs from source encoding 'UTF-8'


✗ One or more pre-commit hooks failed
```

while

```
➜  rails_best_practices .

(ripper):5: regexp encoding option 's' differs from source encoding 'UTF-8'========                                                                 |
Source Code: |======================================================================================================================================|
/home/user/projects/project/app/helpers/business_groups_helper.rb:1 - remove empty helpers
/home/user/projects/project/app/helpers/generic_helper.rb:1 - remove empty helpers
/home/user/projects/project/app/helpers/consumer_center_types_helper.rb:1 - remove empty helpers
/home/user/projects/project/app/views/generic/_form.html.erb:46 - replace instance variable with local variable
/home/user/projects/project/config/routes.rb:3 - restrict auto-generated routes countries (only: [])
/home/user/projects/project/config/routes.rb:4 - restrict auto-generated routes provinces (only: [])
/home/user/projects/project/config/routes.rb:5 - restrict auto-generated routes banking_entities (only: [])
/home/user/projects/project/config/routes.rb:6 - restrict auto-generated routes consumer_center_types (only: [])
/home/user/projects/project/config/routes.rb:7 - restrict auto-generated routes business_groups (only: [])
/home/user/projects/project/config/routes.rb:8 - restrict auto-generated routes users (only: [])

Please go to http://rails-bestpractices.com to see more useful Rails Best Practices.

Found 10 warnings.
```

I'm in overcommit 0.35.0 with rails_best_practices 1.17.0. Do you need any other debugging information?

Thanks

**Comment by sds** [comment]
Would love a pull request addressing this, as we aren't users of the `rails_best_practices` gem at @brigade. @kovpack (who contributed the pre-commit hook) might be able to help here. Thanks!

**Comment by waiting-for-dev** [comment]
Hey, thanks for the response. To be honest I'm not sure I'm going to have the time/motivation to look at this, because currently I'm trying to move away from rails. But, anyway, I wanted to report it just in case it was trivial for people with good knowledge of overcommit codebase. Hopefully someone else will do.

Cheers and thanks for this awesome gem!

**Comment by derwiki** [comment]
@sds OOC why don't ya'll use `rails_best_practices`?

**Comment by derwiki** [comment]
FWIW I thought there was a bug with it, but after [im]patiently waiting, it turns out it took almost 2 minutes to run against a 6 line diff:

```
$ g staged ; time g ci -m "Test"
diff --git a/app/models/invoice_line_item.rb b/app/models/invoice
_line_item.rb
index d691bf3..6d10b20 100644
--- a/app/models/invoice_line_item.rb
+++ b/app/models/invoice_line_item.rb
@@ -61,5 +61,12 @@ module Payment
     def stripe_account
       @stripe_account ||= payment_account.try(:stripe_account)
     end
+
+    def balance_transaction
+      @balance_transaction ||= Stripe::BalanceTransaction
+        .retrieve(bank_transfer['balance_transaction'],
+                  stripe_account: stripe_account_id)
+
+    end
   end
 end
Running pre-commit hooks
Analyze with RailsBestPractices..................[RailsBestPractices] OK

✓ All pre-commit hooks passed

Running commit-msg hooks
Check for trailing periods in subject................[TrailingPeriod] OK
Check text width..........................................[TextWidth] OK
Check subject capitalization.....................[CapitalizedSubject] OK
Check subject line................................[SingleLineSubject] OK

✓ All commit-msg hooks passed

[derwiki/feature 6f1c446] Test
 1 file changed, 7 insertions(+)

real    1m52.338s
user    1m30.629s
sys     0m7.674s
```

**Comment by sds** [comment]
@derwiki I think we considered it at one point, maybe even tried it out but ultimately passed on it.

I just ran it against our codebase, and 95% of the error messages appear to me to be false positives (not necessarily the tool's fault). I think if you used this project from the start it might prove more useful, but we've made some decisions that don't seem to play well with its model of how the world should be, and as a result the tool gets confused about what we're doing.

That's just one perspective. I'd probably consider adopting this for a new Rails project.
